kuwala/pipelines/osm-poi/src/Processor.py

```diff
@@ -308,6 +309,50 @@ def combine_pois(df_node, df_way, df_relation) -> DataFrame:
         df_relation = df_relation.filter(col('is_poi') & col('h3_index').isNotNull()).select(columns)
 
         return df_node.union(df_way).union(df_relation)
+    
+    # search brands and oprerator from names.csv and put that into pyspark dataframe
+
+    @staticmethod
+    def match_brand_and_operator_names(script_dir,spark, df_pois) -> DataFrame:
+            names=pd.read_csv(os.path.join(script_dir, '../tmp/names.csv'))[['display_name','is_operator']]
+            operator_names=names.query('is_operator == 1')['display_name'].tolist()
+            brand_names=names.query('is_operator == 0')['display_name'].tolist()
+
+            operator_names=spark.sparkContext.broadcast(operator_names)
+            brand_names=spark.sparkContext.broadcast(brand_names)
+
+            @udf(returnType=StringType())
+            def brand_name_matching(df_pois_brand):
+                similar_brand_score=-1;best_match_brand=''
+                if(str(df_pois_brand)=='nan'):
+                    best_match_brand=None
+                    return best_match_brand
+                for brand in brand_names:
+                    distance_brand=get_string_distance(df_pois_brand, brand)
+                    if(distance_brand>similar_brand_score):
+                        similar_brand_score=distance_brand
+                        best_match_brand=brand
+                return best_match_brand
+
+            @udf(returnType=StringType())
+            def operator_name_matching(df_pois_operator):
+                similar_operator_score=-1;best_match_operator=''
+                if(str(df_pois_operator)=='nan'):
+                    best_match_operator=None
+                    return best_match_operator
+                for ops in operator_names:
+                    distance_ops=get_string_distance(df_pois_operator,ops)
+                    if(distance_ops>similar_operator_score):
+                        similar_operator_score=distance_ops
+                        best_match_operator=ops
+                return best_match_operator
```

Comment:
Can be put into one method. No distinction necessary between brand and operator names.